plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TestCheckpointEntryIterator.java

```diff
@@ -105,6 +105,24 @@ public void testReadMetadataEntry()
                                 1579190100722L));
     }
 
+    @Test
+    public void testReadProtocolEntries()
+            throws Exception
+    {
+        URI checkpointUri = getResource(TEST_CHECKPOINT).toURI();
```

Comment:
This test is not effective to showcase your fix.

Let's try a different approach:

- expose `getCompletedBytes()` `VisibleForTesting` in `CheckpointEntryIterator`
- use a simplified version of  `io.trino.plugin.deltalake.transactionlog.checkpoint.TestCheckpointWriter#testCheckpointWriteReadJsonRoundtrip` for creating a checkpoint on the fly with one entry of addfile and a lot of removefile entries
- check after getting the addfile entries from the checkpoint entry iterator whether the number of bytes read from the parquet checkpoint file (through previously added `getCompletedBytes()` method) is indeed much smaller than the number of bytes read while reading the remove file entries.
I'm assuming that a non existent entry type in the checkpoint should also imply reading `0` bytes from the parquet file. To be verified.

Reply:
Thank you for the suggestion, I've added a test following this recipe. 

The completedBytes of the Parquet reader includes the bytes used to read the Parquet metadata, hence `txnEntryIterator.getCompletedBytes() != 0`. This also means that we must use a fairly large number of remove entries with random values, such that the size of the remove entries is large relative to the Parquet metadata size.

The test is potentially a bit sensitive to changes to the Parquet writer and reader (e.g., if the writer is modified to write everything as a single row group, or the header becomes very large). For the chosen values the metadata is ~10% of the file size, so the test should not be flaky.

The CheckpointWriter uses the default values for the ParquetWriter. Potentially, these values should be tuned, as the checkpoint files likely have a different access patterns than what is common for Parquet files. It would be beneficial if all non-remove entries are in as few row groups as possible.

`testReadProtocolEntries` was added to have a read test for protocol entries, and to manually test that we had coverage for invalid specific column types. I can remove it if necessary?

Reply:
Take a look at  `CheckpointWriter` 

https://github.com/trinodb/trino/blob/962c0cc7bc4fcd3f20a8fa0765c651ee954ba37c/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java#L127

Now, if you make a slight change on `CheckpointWriter` to provide a  `ParquetWriterOptions` parameter in the constructor, you could customise the size of the row groups and you would probably be able to showcase relatively easy the row group pruning:

```
        ParquetWriterOptions parquetWriterOptions = ParquetWriterOptions.builder()
                .setMaxBlockSize(DataSize.ofBytes(100L))
                .setMaxPageSize(DataSize.ofBytes(100L))
                .build();
```